public/main.js

```diff
@@ -85,6 +76,270 @@ const LibManager = {
 
     return lib;
   },
+
+  acquireModuleMetadata: {},
+
+  /**
+   * @param {string} sourceCode 
+   */
+  detectNewImportsToAcquireTypeFor: async function(sourceCode) {
+
+   /**
+   * @param {string} sourceCode 
+   * @param {string | undefined} mod 
+   * @param {string | undefined} path 
+   */
+    const getTypeDependenciesForSourceCode = async (sourceCode, mod, path) => {
+      // TODO: debounce
+      //
+      // TODO: This needs to be replaced by the AST - it still works in comments 
+      // blocked by https://github.com/microsoft/monaco-typescript/pull/38
+      //
+      // TODO: Support pulling out the root component of a module first to grab that, so it can grab sub-definitions 
+      //
+      // https://regex101.com/r/Jxa3KX/4
+      const requirePattern = /(const|let|var)(.|\n)*? require\(('|")(.*)('|")\);?$/
+      //  https://regex101.com/r/hdEpzO/4
+      const es6Pattern = /(import|export)((?!from)(?!require)(.|\n))*?(from|require\()\s?('|")(.*)('|")\)?;?$/gm
+  
+      const foundModules = new Set()
+      
+      while ((match = es6Pattern.exec(sourceCode)) !== null) {
+        if (match[6]) foundModules.add(match[6])
+      }
+  
+      while ((match = requirePattern.exec(sourceCode)) !== null) {
+        if (match[5]) foundModules.add(match[5])
+      }
+      
+      const moduleJSONURL = (name) => `http://ofcncog2cu-dsn.algolia.net/1/indexes/npm-search/${encodeURIComponent(name)}?x-algolia-agent=Algolia%20for%20vanilla%20JavaScript%20(lite)%203.27.1&x-algolia-application-id=OFCNCOG2CU&x-algolia-api-key=f54e21fa3a2a0160595bb058179bfb1e`
```

Comment:
```suggestion
      const moduleJSONURL = (name) => `http://ofcncog2cu-dsn.algolia.net/1/indexes/npm-search/${encodeURIComponent(name)}?attributes=types&x-algolia-agent=Algolia%20for%20vanilla%20JavaScript%20(lite)%203.27.1&x-algolia-application-id=OFCNCOG2CU&x-algolia-api-key=f54e21fa3a2a0160595bb058179bfb1e`
```

retrieve only the types this way

Reply:
I created a new API Key for TypeScript's usage: `f8eb0884c44049b48f2e4f6688db299b` (note that for significant usage, we'd want a "powered by Algolia" somewhere on the page :)